### PR TITLE
Updating dependencies, smoltcp version

### PR DESCRIPTION
**Pull request by ryan-summers**

### CODE REVIEW

**Comment by ryan-summers** [comment]
bors try

File: Cargo.toml [hunk 1]
```diff
@@ -60,27 +60,21 @@ rev = "70b0eb5"
 # * Utilize `master` branch of smoltcp
 # * Utilize `master` branch of HAL
 [dependencies.stm32h7xx-hal]
-features = ["stm32h743v", "rt", "unproven", "ethernet", "quadspi"]
-# version = "0.9.0"
-git = "https://github.com/quartiq/stm32h7xx-hal.git"
-rev = "33aa67d"
+features = ["stm32h743v", "rt", "unproven", "ethernet", "xspi"]
+git = "https://github.com/stm32-rs/stm32h7xx-hal"
+branch = "v0.11.0"
```
Comment (ryan-summers) [inline Cargo.toml:65]: Let's wait till this gets released, should be pretty soon.

File: Cargo.toml [hunk 2]
```diff
@@ -30,7 +30,7 @@ members = ["ad9959"]
 
 [dependencies]
 cortex-m = { version = "0.7" }
-cortex-m-rt = { version = "0.6", features = ["device"] }
+cortex-m-rt = { version = "0.7", features = ["device"] }
```
Comment (ryan-summers) [inline Cargo.toml:33]: Need to figure out if this is acceptable for removing our patch.

File: Cargo.toml [hunk 3]
```diff
@@ -60,27 +60,21 @@ rev = "70b0eb5"
...
-rev = "a2e3ad5"
-
 [patch.crates-io.heapless]
 git = "https://github.com/quartiq/heapless.git"
 branch = "feature/assume-init"
 
 [dependencies.smoltcp-nal]
 git = "https://github.com/quartiq/smoltcp-nal.git"
-rev = "a28e41f"
+branch = "rs/smoltcp-update"
```
Comment (ryan-summers) [inline Cargo.toml:77]: TODO: Release this if successful

File: src/hardware/adc.rs [hunk 4]
```diff
@@ -149,6 +149,8 @@ macro_rules! adc_input {
     ($name:ident, $index:literal, $trigger_stream:ident, $data_stream:ident, $clear_stream:ident,
      $spi:ident, $trigger_channel:ident, $dma_req:ident, $clear_channel:ident, $dma_clear_req:ident) => {
         paste::paste! {
+            use stm32h7xx_hal::prelude::*;
```
Comment (ryan-summers) [inline src/hardware/adc.rs:152]: Determine why these preludes are needed.

**Comment by ryan-summers** [comment]
bors retry

**Comment by ryan-summers** [comment]
bors retry

**Comment by ryan-summers** [comment]
bors retry

File: Cargo.toml (hunk 1 above)
Comment (jordens) [reply Cargo.toml:65]: Could we land it now anyways so we can collect some running time with the new code?

Comment (ryan-summers) [reply Cargo.toml:65]: Yeah, would be happy to land it. I'll update this to point to a rev.

**Comment by jordens** [comment]
The MAC interrupt fixes are in the hal master now but last I checked not yet in the v0.11 branch. We should also sort that out and remove them here once they are in v0.11.

**Comment by ryan-summers** [comment]
Tracking that in https://github.com/quartiq/stabilizer/issues/479 - as you noted, it's not yet in the v0.11 branch

**Review by jordens — APPROVED** [review]

**Pull request merged**